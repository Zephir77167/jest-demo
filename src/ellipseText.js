//
//  ellipseString
//  jest-demo
//
//  Created on 09/04/2017
//  Copyright (c) 2017 Brigad. All rights reserved.
//

const ELLIPSIS_SYMBOL = '...';

export const ellipseText = (text, maxLength) => {
  if (!text) {
    return '';
  }

  if (!maxLength || maxLength < (1 + ELLIPSIS_SYMBOL.length) || text.length <= maxLength) {
    return text;
  }

  return `${text.substr(0, maxLength - ELLIPSIS_SYMBOL.length)}${ELLIPSIS_SYMBOL}`;
};
