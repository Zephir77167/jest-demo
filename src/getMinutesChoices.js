//
//  getMinutesChoices
//  jest-demo
//
//  Created on 09/04/2017
//  Copyright (c) 2017 Brigad. All rights reserved.
//

export const getMinutesChoices = () => [
  '00',
  '10',
  '20',
  '30',
  '40',
  '50',
];
