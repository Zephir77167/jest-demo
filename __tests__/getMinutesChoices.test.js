//
//  getMinutesChoices.test
//  jest-demo
//
//  Created on 09/04/2017
//  Copyright (c) 2017 Brigad. All rights reserved.
//

import { getMinutesChoices } from '../src/getMinutesChoices';

describe('getMinutesChoices', () => {
  test('only path', () => {
    expect(getMinutesChoices()).toMatchSnapshot();
  });
});
