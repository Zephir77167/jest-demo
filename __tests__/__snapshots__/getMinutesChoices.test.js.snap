// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMinutesChoices only path 1`] = `
Array [
  "00",
  "10",
  "20",
  "30",
  "40",
  "50",
]
`;
